mason.py

```diff
@@ -266,6 +267,13 @@ def get_env_vars(pure_docker_mode: bool, cluster: List[str], beaker_secrets: Lis
                 secret=secret["value"],
             )
         )
+    
+    # Add Git commit and branch information
+    git_commit = subprocess.check_output(["git", "rev-parse", "--short", "HEAD"], text=True).strip()
+    env_vars.append(beaker.EnvVar(name="GIT_COMMIT", value=git_commit))
+    
+    git_branch = subprocess.check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"], text=True).strip()
+    env_vars.append(beaker.EnvVar(name="GIT_BRANCH", value=git_branch))
```

Comment:
does this mean now the tag will be associated with the branch the mason job is run from? I guess usually this is fine, although sometimes im launching stuff from mason to an image from another branch. 

Maybe we could compute the git hash in the dockerfile creation...?